Option view implemented

lukan98 opened the pull request:
Functionality demonstration:
https://user-images.githubusercontent.com/53893982/126509440-69edf945-8c7a-40bd-ae5a-38a713b483c1.mp4

lukan98 committed:
Option view implemented

istaudacher commented on MovieApp/PresentationLayer/HomeScreen/HomeScreenViewController+Design.swift:
```diff
@@ -36,6 +39,12 @@ extension HomeScreenViewController: ConstructViewsProtocol {
+        optionView.snp.makeConstraints {
+            $0.top.equalTo(searchBar.snp.bottom)
+            $0.leading.trailing.equalToSuperview()
+            $0.height.equalTo(40)
```
Extract this to a constant.

istaudacher commented on MovieApp/PresentationLayer/Components/OptionBar/OptionBar.swift:
```diff
@@ -0,0 +1,20 @@
+
+class OptionBar: UIView {
+
+    var currentlySelectedCategory = 0
```
`selectedCategory`

istaudacher commented on MovieApp/PresentationLayer/Components/OptionBar/OptionBar+Design.swift:
```diff
@@ -0,0 +1,100 @@
+    func buildViews() {
+        createViews()
+        styleViews()
+        defineLayoutForViews()
```
remove this line, `defineLayoutForViews()` should be called only from `didMoveToSuperview` in classes that extend `UIView`. So, in OptionBar.swift, override that function and call `defineLayoutForViews()`.

lukan98 replied:
What's the reasoning for this? Intuitively I think it's so that the view is laid out whenever it's position in the superview is changed but I'm not sure.

istaudacher commented on MovieApp/PresentationLayer/Components/OptionBar/OptionBar+Design.swift:
```diff
@@ -0,0 +1,100 @@
+        optionButtonStack.spacing = 22
+
+        for subview in optionButtonStack.arrangedSubviews {
+            guard let button = subview as? UIButton else { return }
```
Did you mean: `continue`?

lukan98 replied:
I felt return was appropriate because if one of the views in the stack wasn't a button the view wouldn't make sense anymore which would mean that something isn't right.

istaudacher commented on MovieApp/PresentationLayer/Components/OptionBar/OptionBar+Design.swift:
```diff
@@ -0,0 +1,100 @@
+    private func onCategorySelection(sender: UIButton) {
+        guard
+            let previouslySelected = optionButtonStack.arrangedSubviews[currentlySelectedCategory] as? UIButton
+        else { return }
```
```
else {
    return
}
```

istaudacher commented on MovieApp/PresentationLayer/Components/OptionBar/OptionBar+Design.swift:
```diff
@@ -0,0 +1,100 @@
+        }
+    }
+
+    @objc
```
Is this annotation necessary?

lukan98 committed:
PR fixes

Pull request issues fixed. SearchBar refactored for composition over inheritance.

lukan98 committed:
super calls added to overriden functions where neccessary
